js/base-util.js

```diff
@@ -214,6 +349,24 @@ function baseUtil () {
 		return JSON.parse(journalFolder);
 	};
 
+	d20plus.ut.charFetchAndRetry = ({char, callback, params = []} = {}) => {
```

Comment:
rewrite as an async method

```js
...
return new Promise(resolve => {
  const wait = setInterval(() => {
    if (attribs.length) {
      resolve(...params)
      clearTimeout(wait)
    }
  }, ...)
})
```

Reply:
Done. The only currently included in this PR usage is still kinda "sync" tho, as the call chain leads to rewriting injected methods